controllers/RatingController.js

```diff
@@ -0,0 +1,131 @@
+import { Op } from 'sequelize';
+import { Favorite, User, Article } from '../database/models';
+import { calculateRating } from '../helpers';
+/**
+ * @author Daniel
+ * @description a class for rating the article
+ */
+class RatingController {
+  /**
+   *
+   * @param {*} req
+   * @param {*} res
+   * @returns {*} object
+   */
+  static async rateArticle(req, res) {
+    const { articleSlug } = req.params;
+    const { rate } = req.body;
+    const { currentUser } = req;
+    const article = await Article.findOne({ where: { slug: articleSlug, status: 'published' } });
+    if (!article) {
+      return res.status(404).send({
+        status: 404,
+        errors: { body: ['Article not found'] }
+      });
+    }
+    const articleId = article.get().id;
+    const ratedArticle = await Favorite.findOne({ where: { articleId, userId: currentUser.id } });
+    if (ratedArticle) {
+      ratedArticle.update({ rating: rate, updatedAt: new Date() });
+      const averageRate = await calculateRating(articleId);
+      return res.status(200).send({
+        status: 200,
+        message: 'Rating updated successfully',
+        article: {
+          ...article.get(),
```

Comment:
@oesukam why?????
In the readme file, on most of the functionalities, they suggest we return the article.

Reply:
@filschristian I don't see the use of returning an article as we did not perform any update on the article. I think we can only return an article in case we have made some update on any of the article's direct information, by direct information, I mean any column of the article's table.

Users will be rating  an article when they already have that particular article in front of them, I think you can only return thing that you need to update users about.